mxcube3/ui/components/SampleQueue/QueueControl.js

```diff
@@ -110,27 +110,34 @@ export default class QueueControl extends React.Component {
   }
 
   render() {
-    const idx = this.props.queue.indexOf(this.props.mounted);
     let nextSample = [];
-
-    if (this.props.queue[idx + 1]) {
-      const sampleData = this.props.sampleList[this.props.queue[idx + 1]];
-      const sampleName = sampleData.sampleName ? sampleData.sampleName : '';
-      const proteinAcronym = sampleData.proteinAcronym ? `${sampleData.proteinAcronym} - ` : '';
-
-      nextSample = [{ text: `Next Sample (${proteinAcronym}${sampleName})`,
+    let idx = -1;
+    if (this.props.queue) {
+      idx = this.props.queue.indexOf(this.props.mounted);
+
+      if (this.props.queue[idx + 1]) {
+        const sampleData = this.props.sampleList[this.props.queue[idx + 1]];
+        const sampleName = sampleData.sampleName ? sampleData.sampleName : '';
+        const proteinAcronym = sampleData.proteinAcronym ? `${sampleData.proteinAcronym} - ` : '';
+
+        nextSample = [{ text: `Next Sample (${proteinAcronym}${sampleName})`,
+                        class: 'btn-default',
+                        action: this.nextSample,
+                        key: 2 }];
+      } else {
+        nextSample = [{ text: 'There is not next sample',
```

Comment:
I'm not entirely sure about this, actually its maybe better to not display the button at all if we cant use it ?, thats what we
thought at least ...